From 97f6b9e1c409569ffb1bdf6b7ec0921e4f0f7a39 Mon Sep 17 00:00:00 2001
From: Dmitry Muhomor <muhomor.dmitry@gmail.com>
Date: Thu, 21 Mar 2024 16:08:35 +0200
Subject: [PATCH 06/11] Revert "[Invisalign2] Encapsulate
 tBTM_SEC_CB::RemoveServiceById"

This reverts commit e06e22debc40ec2e2c8cf725c8c9dff8cd65e535.
---
 system/stack/btm/btm_sec.cc    | 17 ++++++++++++++++-
 system/stack/btm/btm_sec_cb.cc | 19 -------------------
 system/stack/btm/btm_sec_cb.h  |  1 -
 3 files changed, 16 insertions(+), 21 deletions(-)

diff --git a/system/stack/btm/btm_sec.cc b/system/stack/btm/btm_sec.cc
index 793f0eddd9..d162c13a97 100644
--- a/system/stack/btm/btm_sec.cc
+++ b/system/stack/btm/btm_sec.cc
@@ -450,7 +450,22 @@ bool BTM_SetSecurityLevel(bool is_originator, const char* p_name,
  *
  ******************************************************************************/
 uint8_t BTM_SecClrService(uint8_t service_id) {
-  return btm_sec_cb.RemoveServiceById(service_id);
+  tBTM_SEC_SERV_REC* p_srec = &btm_sec_cb.sec_serv_rec[0];
+  uint8_t num_freed = 0;
+  int i;
+
+  for (i = 0; i < BTM_SEC_MAX_SERVICE_RECORDS; i++, p_srec++) {
+    /* Delete services with specified name (if in use and not SDP) */
+    if ((p_srec->security_flags & BTM_SEC_IN_USE) &&
+        (p_srec->psm != BT_PSM_SDP) &&
+        (!service_id || (service_id == p_srec->service_id))) {
+      LOG_VERBOSE("BTM_SEC_CLR[%d]: id %d", i, service_id);
+      p_srec->security_flags = 0;
+      num_freed++;
+    }
+  }
+
+  return (num_freed);
 }
 
 /*******************************************************************************
diff --git a/system/stack/btm/btm_sec_cb.cc b/system/stack/btm/btm_sec_cb.cc
index d3e0c6f388..1ecdaa9ce2 100644
--- a/system/stack/btm/btm_sec_cb.cc
+++ b/system/stack/btm/btm_sec_cb.cc
@@ -28,7 +28,6 @@
 #include "osi/include/list.h"
 #include "stack/btm/btm_dev.h"
 #include "stack/btm/security_device_record.h"
-#include "stack/include/bt_psm_types.h"
 #include "types/raw_address.h"
 
 void tBTM_SEC_CB::Init(uint8_t initial_security_mode) {
@@ -305,21 +304,3 @@ bool tBTM_SEC_CB::AddService(bool is_originator, const char* p_name,
 
   return (record_allocated);
 }
-
-uint8_t tBTM_SEC_CB::RemoveServiceById(uint8_t service_id) {
-  tBTM_SEC_SERV_REC* p_srec = &sec_serv_rec[0];
-  uint8_t num_freed = 0;
-  int i;
-
-  for (i = 0; i < BTM_SEC_MAX_SERVICE_RECORDS; i++, p_srec++) {
-    /* Delete services with specified name (if in use and not SDP) */
-    if ((p_srec->security_flags & BTM_SEC_IN_USE) &&
-        (p_srec->psm != BT_PSM_SDP) &&
-        (!service_id || (service_id == p_srec->service_id))) {
-      LOG_VERBOSE("BTM_SEC_CLR[%d]: id:%d", i, service_id);
-      p_srec->security_flags = 0;
-      num_freed++;
-    }
-  }
-  return (num_freed);
-}
diff --git a/system/stack/btm/btm_sec_cb.h b/system/stack/btm/btm_sec_cb.h
index 8ed832c889..2482ccdd55 100644
--- a/system/stack/btm/btm_sec_cb.h
+++ b/system/stack/btm/btm_sec_cb.h
@@ -96,7 +96,6 @@ class tBTM_SEC_CB {
   bool AddService(bool is_originator, const char* p_name, uint8_t service_id,
                   uint16_t sec_level, uint16_t psm, uint32_t mx_proto_id,
                   uint32_t mx_chan_id);
-  uint8_t RemoveServiceById(uint8_t service_id);
 };
 
 extern tBTM_SEC_CB btm_sec_cb;
-- 
2.34.1

